src/video_core/command_classes/codecs/codec.h

```diff
@@ -0,0 +1,56 @@
+// Copyright 2020 yuzu Emulator Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+
+#pragma once
+#include <memory>
+#include <vector>
+#include "common/common_funcs.h"
+#include "common/common_types.h"
+#include "video_core/command_classes/nvdec_common.h"
+
+extern "C" {
+#include <libavcodec/avcodec.h>
+}
```

Comment:
```suggestion
extern "C" {
#ifndef _WIN32
#pragma GCC diagnostic ignored "-Wconversion"
#endif
#include <libavcodec/avcodec.h>
#ifndef _WIN32
#pragma GCC diagnostic pop
#endif
}
```

Prevents the compiler from throwing `-Werror=conversion` errors while reading FFmpeg headers.

Reply:
Shouldn't we test for GCC usage instead of target platform? GCC can be used on windows as well.

Reply:
Do you happen to know the variable that would test for GCC? Throw me a bone, here

Reply:
`__GNUC__` macro should do the job here

Reply:
I'd suggest testing for GCC or Clang with

```C++
#if defined(__GNUC__) || defined(__clang__)
```

Both of them support `#pragma GCC`

Reply:
Thanks, it's updated

Reply:
There's sneaky extra space in first `#if`

Reply:
@lat9nq this is addressed in the latest commit. Please let me know if there are still build issues in Fedora.